src/rdkafka.c

```diff
@@ -346,6 +346,29 @@ rd_kafka_resp_err_t rd_kafka_oauthbearer_set_token_failure(rd_kafka_t *rk,
 #endif
 }
 
+rd_kafka_error_t *rd_kafka_set_sasl_creds(rd_kafka_t *rk, const char *username, const char *password) {
+        char errstr[256];
+        rd_kafka_conf_t *conf = &rk->rk_conf; /* Not using rd_kafka_conf because of const qualifier */
```

Comment:
I'm not sure I understand this comment, rk_conf is non-const.

Reply:
const rd_kafka_conf_t *rd_kafka_conf(rd_kafka_t *rk) returns a const* to an rd_kafka_conf_t. The calls to lock/unlock break that const modifier.

Reply:
No need for rd_kafka_conf(), that's part of the public API. Inside librdkafka you should access rk->rk_conf directly.